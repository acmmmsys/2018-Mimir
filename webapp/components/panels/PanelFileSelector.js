import React from 'react';
import { connect } from 'react-redux';
import { compose, withProps, nest, lifecycle } from 'recompose';
import { selectFile, requestImages, requestImageVisualization, requestCnnClassification } from 'actions';

import { isEmpty } from 'utils';

import Panel from 'layout/Panel'
import ImageDropzone from 'components/ui/ImageDropzone';
import FileSelector from 'components/ui/FileSelector';



const enhance = compose(
    connect(
        state => ({
            files: state.app.images,
            loading: state.loading.files,
            selectedFile: state.cnn.selectedImageId,
            selectedLayer: state.cnn.selectedLayer,
            selectedClass: state.cnn.selectedClass,
        }),
        dispatch => ({
            requestImages() {
                dispatch(requestImages())
            },
            selectFile(imageId, layerId, classId) {
                dispatch(selectFile(imageId))
                dispatch(requestCnnClassification(imageId))
                dispatch(requestImageVisualization(imageId, layerId, classId))
            },
        }),
        (state, dispatch, ownProps) => ({
            ...ownProps,
            ...state,
            ...dispatch,
            selectFile(imageId) {
                dispatch.selectFile(imageId, state.selectedLayer, state.selectedClass)
            }
        })
    ),
    lifecycle({
        componentWillMount() {
            this.props.requestImages();
        }
    }),
    withProps(props => ({
        isEmpty: isEmpty(props.files),
        emptyMessage: ( <ImageDropzone /> ),
    }))
);

export default enhance(nest(Panel, FileSelector));