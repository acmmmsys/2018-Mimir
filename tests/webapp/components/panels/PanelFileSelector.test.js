import React from 'react';
import PanelFileSelector from '../../../../webapp/components/panels/PanelFileSelector'
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store'
 
const mockStore = configureStore();

let store;
let wrapper;

test('PanelFileSelector does render', () => {
  const wrapper = shallow(
      <PanelFileSelector store={mockStore(initialState)}>This is a test!</PanelFileSelector>
  );
  expect(wrapper).toMatchSnapshot();
});
