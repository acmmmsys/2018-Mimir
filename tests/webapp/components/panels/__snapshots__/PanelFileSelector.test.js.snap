// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PanelFileSelector does render 1`] = `
<lifecycle(withProps(nest(defaultProps(branch(branch(Panel))), FileSelector)))
  files={Array []}
  loading={false}
  requestImages={[Function]}
  selectFile={[Function]}
  selectedClass={6}
  selectedLayer="block5_conv4"
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
>
  This is a test!
</lifecycle(withProps(nest(defaultProps(branch(branch(Panel))), FileSelector)))>
`;
